tests-beta/android/MASVS-NETWORK/MASTG-TEST-0x22-1.md

```diff
@@ -0,0 +1,38 @@
+---
+title: Missing Certificate Pinning in Code
+platform: android
+id: MASTG-TEST-0x22-1
+type: [static]
+weakness: MASWE-0047
+---
+
+## Overview
+
+Apps can configure certificate pinning using the [Network Security Configuration]("../../../Document/0x05g-Testing-Network-Communication.md#certificate-pinning"). For each domain, one or multiple digests can be pinned.
+
+Certificate pinning can also be done manually in the code. Depending on the used technologies, this can be done for example by:
+
+- Pinning a certificate with a custom `TrustManager`,
+- configuring the used third party networking libraries to pin certificates,
+- use plugins to achieve certificate pinning for hybrid apps.
+
+Chapter [Certificate pinning without Android Network Security Configuration]("../../../Document/0x05g-Testing-Network-Communication.md#certificate-pinning-without-android-network-security-configuration") explains in more detail how this can be achieved in the app.
+
+The goal of this test is to check if any certificate pinning exists.
+
+!!! note "Limitations"
+    Since there are many different ways to achieve certificate pinning in the code, checking statically if the application performs pinning might not reveal all such locations. To make sure certificates are pinned for all relevant connections, additional dynamic analysis can be performed.
```

Comment:
```suggestion
    Since there are many different ways to achieve certificate pinning in the code, checking statically if the application performs pinning might not reveal all such locations. To make sure certificates are pinned for all relevant connections, additional dynamic analysis need to be performed.
```

Reply:
Dynamic analysis is a must to effectively test Certificate Pinning.